selftest/map-getmapsbyname/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import "C"
+
+import (
+	"log"
+	"syscall"
+	"unsafe"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+)
+
+const (
+	BPFMapNameToNotFind = "not_found"
+	// The following properties are used to identify the map
+	BPFMapNameToFind       = "test_name"
+	BPFMapTypeToFind       = bpf.MapTypeArray
+	BPFMapMaxEntriesToFind = 1
+	BPFMapKeySizeToFind    = 4
+	BPFMapValSizeToFind    = 4
+)
+
+func main() {
+	bpfModule, err := bpf.NewModuleFromFile("main.bpf.o")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer bpfModule.Close()
+
+	bpfModule.BPFLoadObject()
+
+	notFoundMapsIDs, err := bpf.GetMapsIDsByName(BPFMapNameToNotFind)
+	if len(notFoundMapsIDs) != 0 {
+		log.Fatalf("the %s map should not be found, but it was found with ids: %v", BPFMapNameToNotFind, notFoundMapsIDs)
+	}
+
+	mapsIDs, err := bpf.GetMapsIDsByName(BPFMapNameToFind)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if len(mapsIDs) == 0 {
+		log.Fatalf("the %s map was not found", BPFMapNameToFind)
+	}
+
+	// try to identify the map by its properties
+	similarMaps := []*bpf.BPFMapLow{}
+	for _, id := range mapsIDs {
+		bpfMap, err := bpf.GetMapByID(id)
+		if err != nil {
+			log.Fatalf("the %s map with %d id was not found: %v", BPFMapNameToFind, id, err)
+		}
+
+		if bpfMap.Type() == BPFMapTypeToFind &&
```

Comment:
Yeah, flags can be more one field to check. We only need to implement it:

![image](https://github.com/aquasecurity/libbpfgo/assets/3372117/df9554ea-aba2-400d-b3f8-b1010dfedd06)

Reply:
It will help a lot to have flags because I will need it for our project to identify the origin of maps 👍 
I can take a look to implement this feature if it could help 😄

Reply:
> I can take a look to implement this feature if it could help 😄

Awesome! ❤️